### PR TITLE
Feature/add other users

### PR DESCRIPTION
**Pull request by hammad-Ikhlaq-7vals**
Invitations to other users

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -134,6 +134,55 @@
...
+  # The column name used for counter_cache column. If this is nil,
+  # the #invited_by association is declared without counter_cache.
+  # Default: nil
+  # config.invited_by_counter_cache = :invitations_count
+
+  # Auto-login after the user accepts the invite. If this is false,
+  # the user will need to manually log in after accepting the invite.
+  # Default: true
+  # config.allow_insecure_sign_in_after_accept = false
+
```
Comment (mujtaba-saboor) [inline config/initializers/devise.rb:185]: enable `allow_unconfirmed_access_for = 2.days`

File: db/migrate/20200922054158_add_invitable_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      add_column :users, :invitation_sent_at, :datetime
+      add_column :users, :invitation_accepted_at, :datetime
+      add_column :users, :invitation_limit, :integer
+      add_column :users, :invited_by_id, :integer
+      add_column :users, :invited_by_type, :string
+      add_index :users, :invitation_token, unique: true
+  end
+
+  def up
+  change_column :users, :invitation_token, :string, limit: nil
```
Comment (mujtaba-saboor) [inline db/migrate/20200922054158_add_invitable_to_users.rb]: indent

File: db/migrate/20200922063521_add_default_to_user_column_company_id.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddDefaultToUserColumnCompanyId < ActiveRecord::Migration[6.0]
+  def change
+    change_column_default :users, :company_id, 1
```
Comment (mujtaba-saboor) [inline db/migrate/20200922063521_add_default_to_user_column_company_id.rb]: why?

File: db/schema.rb [hunk 4]
```diff
@@ -10,9 +10,30 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_09_09_083144) do
+ActiveRecord::Schema.define(version: 2020_09_22_071405) do
 
-  create_table "comments", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci", force: :cascade do |t|
```
Comment (mujtaba-saboor) [inline db/schema.rb:15]: this shouldnt be pushed, discuss with mujeeb for gitignore.

**Comment by mujtaba-saboor** [comment]
cycle 1

File: config/initializers/devise.rb (hunk 1 above)
Comment (hammad-Ikhlaq-7vals) [reply config/initializers/devise.rb:185]: done

File: db/migrate/20200922054158_add_invitable_to_users.rb (hunk 2 above)
Comment (hammad-Ikhlaq-7vals) [reply db/migrate/20200922054158_add_invitable_to_users.rb]: done

File: db/schema.rb (hunk 4 above)
Comment (hammad-Ikhlaq-7vals) [reply db/schema.rb:15]: ok

File: db/migrate/20200922063521_add_default_to_user_column_company_id.rb (hunk 3 above)
Comment (hammad-Ikhlaq-7vals) [reply db/migrate/20200922063521_add_default_to_user_column_company_id.rb]: multitenancy has not been merged yet. Once it is merged, I will get this from subdomain.

Comment (mujtaba-saboor) [reply db/migrate/20200922063521_add_default_to_user_column_company_id.rb]: remove this migration, we don't need this

**Pull request merged**